### DISCUSSION TITLE
Temperature dependent battery degradation

### DISCUSSION THREAD

**Discussion by zafar594** [pybamm-team/PyBaMM#3917] [General]
I wanted to analyze battery degradation due to variation in ambient temperatures, is there a degradation model that takes into account temperature? If yes, then which option(s) should I choose to make this degradation function?

**Comment by abhishek-appana** [comment]
It depends on which degradation model you would like to implement! 

For example, SEI degradation submodel (File name: sei_growth.py) takes into account the temperature variation. You can select one of the following SEI options for this:
- "reaction limited"
- "interstitial-diffusion limited"
- "ec reaction limited"

**Reply by zafar594** [reply]
Thanks for answering! Do you know where I can read more about these SEI options, to decide which one to select under what conditions, for example?

**Reply by abhishek-appana** [reply]
PyBaMM has all the documentation for SEI on this [webpage](https://docs.pybamm.org/en/stable/source/api/models/submodels/interface/sei/index.html). You can read the reference papers for further understanding!

The other option is to go through the code. The codes for all the SEI models are [here](https://github.com/pybamm-team/PyBaMM/tree/develop/pybamm/models/submodels/interface/sei).

**Reply by zafar594** [reply]
Thank you!

**Reply by zafar594** [reply]
In my case I have a numpy array for temperature values which I am trying to use as ambient temperature, updated at every time step. I am using this for reference [ambi_temp](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/scripts/DFN_ambient_temperature.py) but in my case, instead of a function of time, temperature is simply an array. I tried to modify the input like this (temps is an array for temperature values and duration is length of experiment):

```
def amb_temp(y, z, t):
    for t in range(duration):
        return temp[t]

```
but it only takes the first value (temp[0]) for all data points. Otherwise, defining it like this:

```
def amb_temp(y, z, t):
    return temp[t]
```

doesn't work at all. What am I doing wrong? Please help!

**Reply by Sunnyturicum** [reply]
Hi
Not sure which cell you are using for your analysis. But I have done some temperature related work with Pybamm here: https://github.com/ArnaudOlt/dissertation_pybamm
Maybe it helps.
